crates/partition-store/src/partition_store_manager.rs

```diff
@@ -220,6 +226,25 @@ impl PartitionStoreManager {
             .cloned()
             .ok_or(SnapshotError::PartitionNotFound(partition_id))?;
 
+        if let Some(min_target_lsn) = min_target_lsn {
+            let applied_lsn = partition_store
+                .get_applied_lsn()
+                .await
+                .map_err(|e| SnapshotError::SnapshotExport(partition_id, anyhow!(e)))?
+                .ok_or(SnapshotError::SnapshotExport(
+                    partition_id,
+                    anyhow!(StorageError::DataIntegrityError),
+                ))?;
```

Comment:
create_snapshot() does the same db read. Is there a value from doing this check here and not moving it into  `PartitionStore::create_snapshot` and let it create the directory after it runs this check?

Reply:
Done now!

I am not very happy with the way error handling turned out; I also tried introducing an error type specific to `create_snapshot` but that was even worse. LMK what you think - am I missing some error handling trick there?